### PR TITLE
SIMSBIOHUB-591: Show all user's data on Project page & add search functionality

### PR DESCRIPTION
**Pull request by mauberti-bc**
### Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-591

Goal is to display all Projects, Surveys and Observations that the user has access to on the Project page, and improve searching on this information

###  Description of Changes

Frontend:
Display all Projects and Surveys that the user has access to
Display all Observations, Telemetry, and Animals that the user has access to
Added the ability to search for Surveys, Observations, Telemetry, and Animals
Removed advanced filters

Backend:
Updated the search logic in the project-repository
Added search logic to the survey-repository and service
Added search logic to the observation-repository and service

### CODE REVIEW

File: app/src/constants/misc.ts [hunk 1]
```diff
@@ -13,8 +12,8 @@ export enum AdministrativeActivityStatusType {
   REJECTED = 'Rejected'
 }
 
-export const SurveyProgressChipColours: Record<string, Color> = {
-  PLANNING: blue,
-  'IN PROGRESS': purple,
-  COMPLETED: green
-};
+export const SurveyProgressChipColours = [
```
Comment (MacDeluca) [inline app/src/constants/misc.ts]: This should still be a record.

File: app/src/constants/taxon.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { blue, deepOrange, grey, indigo, pink, purple, teal } from '@mui/material/colors';
 
 export const taxonRankColours = [
```
Comment (MacDeluca) [inline app/src/constants/taxon.ts]: This should be a record.

File: app/src/features/summary/list-data/project/ProjectsListContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  const codesContext = useCodesContext();
+  const taxonomyContext = useTaxonomyContext();
+
+  const { searchParams, setSearchParams } = useSearchParams<StringValues<ProjectDataTableURLParams>>();
+
+  useEffect(() => {
+    codesContext.codesDataLoader.load();
+  }, [codesContext.codesDataLoader]);
+
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({
```
Comment (MacDeluca) [inline app/src/features/summary/list-data/project/ProjectsListContainer.tsx:76]: I think some comments here around the setStates would be helpful and maybe why we prepend 'p_' on the properties. (I think it was something to do with multiple tables being rendered).

**Comment by MacDeluca** [comment]
It appears the 'Species' search is doing some strange formatting when typing.

Searching 'alces' and clicking 'Alces' replaces the search with Moose (I think this is intentional).
Searching 'alces alces' replaces the text with italics.

Everything else looks good to me!

**Review by MacDeluca — DISMISSED** [review]
🌡️

File: app/src/features/summary/list-data/project/ProjectsListContainer.tsx (hunk 3 above)
Comment (NickPhura) [reply app/src/features/summary/list-data/project/ProjectsListContainer.tsx:76]: added!

**Review by NickPhura — APPROVED** [review]
👍

**Pull request merged**